src/app/scopes-dialog/scopes-dialog.component.html

```diff
@@ -1,28 +1,31 @@
-<div class="ms-Dialog center-dialog ms-Dialog--close" id="scopes-dialog">
-    <button class="ms-Dialog-button ms-Dialog-buttonClose">
+<div class="ms-Dialog center-dialog ms-Dialog--close" id="scopes-dialog" tabindex="0">
+    <!--
+    This empty tag is a workaround for fixing an off by one error.
+
+    When focus is on the CLOSE button on the bottom right of the dialog and the user presses TAB focus would skip to the dialog title.
+
+    Having this empty tag means that focus would skip to the CLOSE icon instead of the dialog title.
+    -->
+    <div #firstElement tabindex="0"></div>
+    <button class="ms-Dialog-button ms-Dialog-buttonClose" tabindex="0" (keydown)="focusOnLastElement($event, lastElement)"
+        autofocus>
         <i class="ms-Icon ms-Icon--Cancel"></i>
     </button>
-    <div class="ms-Dialog-title">{{getStr('modify permissions')}}</div>
-    <p class="ms-Dialog-subText">Select different
+    <div tabindex="0" class="ms-Dialog-title">{{getStr('modify permissions')}}</div>
+    <p class="ms-Dialog-subText" tabindex="0">Select different
         <a class="ms-Link" href="https://developer.microsoft.com/graph/docs/concepts/permissions_reference" target="_blank">permissions</a>
         to try out Microsoft Graph API endpoints.</p>
     <div class="ms-Dialog-content">
-
-        <div id="scopes-list-table-container" #scopesListTableContainer [style.height.px]="getScopesListTableHeight()">
+        <div id="scopes-list-table-container" #scopesListTableContainer [style.height.px]="getScopesListTableHeight()"
+            tabindex="0">
             <table class="ms-Table" id="scopes-list-table">
                 <tr *ngFor="let scope of scopes">
-                    <td class="scopeRow">
+                    <td class="scopeRow" [tabindex]="getTabIndex(scope)">
                         <div class="c-checkbox">
                             <label class="c-label">
                                 <!-- scope.consented indicates that the user or admin has previously consented to the scope.-->
-                                <input type="checkbox" 
-                                    [disabled]="scope.consented"
-                                    (change)="toggleRequestScope(scope)"
-                                    name="checkboxId1" 
-                                    value="value1" 
-                                    [attr.aria-label]="getScopeLabel(scope.name)"
-                                    [tabindex]=""
-                                    >
+                                <input type="checkbox" [disabled]="scope.consented" (change)="toggleRequestScope(scope)"
```

Comment:
The checkbox isn't accessible. We can't check the checkbox. Please make this accessible.

Reply:
How do you check a checkbox in a browser with the keyboard?  In Windows it was space, but currently space just scrolls down.

Reply:
@MIchaelMainer  I am able to check the checkbox from the keyboard. Are using space or enter? You should use space.

@darrelmiller It scrolls if focus is not on the checkbox.

Best way to test this is to have narrator on.

Reply:
I see what is going on. The checkbox does get focus and I can select with the keyboard. First tab gives the table cell focus which is used to provide the crappy onhover indicator I added. The next tab is when the checkbox gets focus. We just don't have a visible indicator that the checkbox got focus. 

Ideally, we only tab through the checkboxes (the table cell doesn't get a tabindex), but the table cell still gets onhover indicator.